app/workers/functions.php

```diff
@@ -210,6 +204,47 @@ public function perform()
                  *      On failure add error count
                  *      If error count bigger than allowed change status to pause
                  */
+
+                // Reschedule
+                Authorization::disable();
+                $function = $database->getDocument($functionId);
+                Authorization::reset();
+
+                if (empty($function->getId()) || Database::SYSTEM_COLLECTION_FUNCTIONS != $function->getCollection()) {
+                    throw new Exception('Function not found ('.$functionId.')');
+                }
+
+                if($scheduleOriginal && $scheduleOriginal !== $function->getAttribute('schedule')) { // Schedule has changed from previous run, ignore this run.
+                    return;
+                }
+
+                $cron = CronExpression::factory($function->getAttribute('schedule'));
+                $next = (int) $cron->getNextRunDate()->format('U');
+
+                $function
+                    ->setAttribute('scheduleNext', $next)
+                    ->setAttribute('schedulePrevious', \time())
+                ;
+
+                Authorization::disable();
+
+                $function = $database->updateDocument(array_merge($function->getArrayCopy(), [
+                    'scheduleNext' => $next,
+                ]));
+
+                Authorization::reset();
+
+                ResqueScheduler::enqueueAt($next, 'v1-functions', 'FunctionsV1', [
+                    'projectId' => $projectId,
+                    'functionId' => $function->getId(),
+                    'executionId' => null,
+                    'trigger' => 'schedule',
+                    'scheduleOriginal' => $function->getAttribute('schedule', ''),
+                ]);  // Async task rescheduale
+
+                Swoole\Coroutine\run(function () use ($trigger, $projectId, $executionId, $database, $function) {
+                    $this->execute($trigger, $projectId, $executionId, $database, $function);
```

Comment:
Is there no `go()` from Swoole necessary?

Reply:
Umm... I'll give it a check

Reply:
Removed co-routines for now.